python/pypto/runtime/execute_artifact.py

```
@@ -0,0 +1,387 @@
+    (``PYPTO_EXEC_RESULT=INFRA``) rather than a device test failure
+    (``=FAIL``), so a cache/setup miss isn't misread as a numerical regression.
+    """
+
+
+def execute_artifact_dir(
+    work_dir: Path,
+    platform: str,
+    device_id: int,
+    *,
+    pto_isa_commit: str | None = None,
+    dfx: _DfxOpts = _DfxOpts(),
```

Ruff (0.15.20)

[warning] Do not perform function call `_DfxOpts` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)